infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/OpenShiftInfrastructure.java

```diff
@@ -62,13 +71,37 @@ public OpenShiftInfrastructure(
     this.runtimeContextFactory = runtimeContextFactory;
     this.dockerImageEnvConverter = dockerImageEnvConverter;
     this.runtimeStatusesCache = runtimeStatusesCache;
+    this.entityManager = entityManager;
   }
 
   @Override
   public Set<RuntimeIdentity> getIdentities() throws InfrastructureException {
     return runtimeStatusesCache.getIdentities();
   }
 
+  @Override
+  @Transactional(rollbackOn = ServerException.class)
+  public Map<String, WorkspaceStatus> findWorkspaceActivityDiscrepancies() throws ServerException {
```

Comment:
Infrastructure does not seem to be a thing that should do DB calls. Let me try to understand a cause and propose some alternatives ways

Reply:
`KubernetesRuntimeState`? The kubernetes runtime definitely does do DB calls. Maybe not directly in the infrastructure but the classes it depends on, do access the DB.

Reply:
We'll not use this implementation.